iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/helpers/IntegrationTest.java

```diff
@@ -67,4 +70,6 @@ protected void finished(Description description)
     public static boolean runIotHubTests = Boolean.parseBoolean(Tools.retrieveEnvironmentVariableValue("RUN_IOTHUB_TESTS", "true"));
     public static boolean runProvisioningTests = Boolean.parseBoolean(Tools.retrieveEnvironmentVariableValue("RUN_PROVISIONING_TESTS", "true"));
 
+    // Infinite read timeout for all http operations
+    public static int HTTP_READ_TIMEOUT = 0;
```

Comment:
Note that only the read timeout will be infinite. The tests will still timeout if they fail to connect to iothub. With this change, they just won't throw an exception if iothub takes a while to finish the operation and return the http response